src/command/tools/merge.rs

```diff
@@ -0,0 +1,114 @@
+use apollo_parser::ast;
+use apollo_parser::Parser as ApolloParser;
+use clap::Parser;
+use serde::Serialize;
+
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::ffi::OsStr;
+
+use crate::options::ToolsMergeOpt;
+use crate::{RoverOutput, RoverResult};
+
+#[derive(Clone, Debug, Parser, Serialize)]
+pub struct Merge {
+    #[clap(flatten)]
+    options: ToolsMergeOpt,
+}
+
+impl Merge {
+    pub fn run(&self) -> RoverResult<RoverOutput> {
+        // find files by extension
+        let schemas = self.find_files_by_extensions(self.options.schemas.clone(), &["graphql", "gql"])?;
+        // merge schemas into one
+        let schema = self.merge_schemas_into_one(schemas)?;
+        Ok(RoverOutput::ToolsSchemaMerge(schema))
+    }
+
+    fn find_files_by_extensions<P: AsRef<Path>>(&self, folder: P, extensions: &'_ [&str]) -> std::io::Result<Vec<PathBuf>> {
+        let mut result = Vec::new();
+        for entry in fs::read_dir(folder.as_ref())? {
+            let entry = entry?;
+            let path = entry.path();
+            if path.is_dir() {
+                let subfolder_result = self.find_files_by_extensions(&path, extensions);
+                if let Ok(subfolder_paths) = subfolder_result {
+                    result.extend(subfolder_paths);
+                }
+            } else if let Some(file_ext) = path.extension().and_then(OsStr::to_str) {
+                if extensions.contains(&file_ext) {
+                    result.push(path);
+                }
+            }
+        }
+    
+        Ok(result)
+    }
+
+    fn merge_schemas_into_one(&self, schemas: Vec<PathBuf>) -> RoverResult<String> {
+        let mut schema = apollo_encoder::Document::new();
+        for schema_path in schemas {
+            let schema_content = fs::read_to_string(schema_path)?;
+            let parser = ApolloParser::new(&schema_content);
+            let ast = parser.parse();
+            let doc = ast.document();
+            
+            for def in doc.definitions() {
```

Comment:
Dumb question- does apollo-parser automatically aggregate multiple definitions together? Meaning, if I have: 

```gql
type User @key(fields:"id") {
id: ID!
name: String!
} 
```

and 

```gql
type User @key(fields:"id") {
id: ID!
username: String!
} 
```

Would I get a combo of both, an error, or one of them?

Reply:
A quick test suggests you will get both of them.

Reply:
Amazing, thanks for testing!